### PR TITLE
[BEAM-3126] Adding a new Flatten test to Python SDK.

### PR DESCRIPTION
**Pull request by youngoli**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

In order to help test the new flatten operation added to the Python bundle processor, I'm adding a ValidatesRunner test based on an existing test in FlattenTest.java (testFlattenMultiplePCollectionsHavingMultipleConsumers).

### CODE REVIEW

**Comment by youngoli** [comment]
Run Java PreCommit

**Comment by youngoli** [comment]
Run Java Gradle PreCommit

**Comment by youngoli** [comment]
R: @robertwb

File: sdks/python/apache_beam/transforms/ptransform_test.py [hunk 1]
```diff
@@ -535,6 +535,29 @@ def test_flatten_input_type_must_be_iterable_of_pcolls(self):
...
+    input = pipeline | 'Start' >> beam.Create(['AA', 'BBB', 'CC'])
+
+    class SomeDoFn(beam.DoFn):
+      def process(self, element):
+        if len(element) % 2 == 0:
+          yield element
+        else:
+          yield pvalue.TaggedOutput('odd_length', element)
+
+    results = (input | 'Split' >> beam.ParDo(SomeDoFn())
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/ptransform_test.py]: I might write this as

```
def split_even_odd(element):
  tag = 'even' if len(element) % 2 == 0 else 'odd'
  return pvalue.TaggedOutput(tag, element)
even_length, odd_length = input | beam.Map(even_odd).with_outputs('even', 'odd')
```

Comment (youngoli) [reply sdks/python/apache_beam/transforms/ptransform_test.py]: Done.

**Pull request merged**